src/core/string_set.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <optional>
+
+#include "dense_set.h"
+
+extern "C" {
+#include "redis/sds.h"
+}
+
+namespace dfly {
+
+class StringSet : public DenseSet {
+ public:
+  uint64_t Hash(const void* ptr) const override;
+
+  bool Equal(const void* ptr1, const void* ptr2) const override;
+
+  size_t ObjectAllocSize(const void* s1) const override;
+
+  bool Add(std::string_view s1);
+
+  bool Add(sds s1);
```

Comment:
it's pretty dangerous to have the same method that accepts `sds` (char*) and `string_view` that has an implicit casting from `char*`.  I noticed how it forced you to pass explicit types when calling Add. 

1. Why do you need both interfaces? Do you actually use both?
2. Consider introducing methods with different names. 

In general, C++ function overriding is a messy beast for many reasons. Golang and Rust do not allow it, for example.

Reply:
I ended up introducing both because during a load from save file wrapping an SDS in a string_view caused an allocation and copy of an already allocated string which was causing a large gap between the load times of `DenseSet` and the original dictionary type.